File: max-consecutive-ones-ii/max-consecutive-ones-ii.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
       
        int left=0,right,k=1;
        for( right =0;right<nums.size();right++){
            if(nums[right]==0)
                k--;
            if(k<0){
                if(nums[left]==0)k++;
                    left++;
                
            }
        }
        return right-left;
    }
};
        

File: 487-max-consecutive-ones-ii/max-consecutive-ones-ii.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        left = 0
        right = 0
        ans = 0
        sum = 0
        for right in range(len(nums)):
            if nums[right] == 0:
                sum += 1
            while sum>1:
                if nums[left] == 0:
                    sum-=1
                left += 1
            ans = max(ans, right-left+1)
        return ans